src/flamegraph/color/palettes.rs

```diff
@@ -1,3 +1,29 @@
+pub(super) mod annotated {
+    use crate::flamegraph::color::BasicPalette;
+
+    /// Handle annotations (_[j], _[i], ...; which are
+    /// accurate) in a generic way.
+    pub fn resolve(name: &str) -> BasicPalette {
+        if name.ends_with(']') {
```

Comment:
Neat! But since this code is the same as the Java code, maybe move it out into a shared helper function that returns `Option<BasicPalette>`?

Reply:
That looks a lot better, thanks!